lib/puppet/type/cisco_fabricpath_topology.rb

```diff
@@ -0,0 +1,112 @@
+# Manages a Cisco fabricpath Topology.
+#
+# Sep 2015
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ipaddr'
+begin
+  require 'puppet_x/cisco/cmnutils'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..',
+                                     'puppet_x', 'cisco', 'cmnutils.rb'))
+end
+
+Puppet::Type.newtype(:cisco_fabricpath_topology) do
+  @doc = "Manages a Cisco fabricpath Topology.
+
+  cisco_fabricpath_topology {'<topo_id>':
+    ..attributes..
+  }
+
+  <topo_id> is the id of the topology.
+
+  Example:
+    cisco_fabricpath_topology {'1':
+      ensure       => present,
+      topo_name    => 'Topo-1',
+      member_vlans => ['101-200', '250']
+      member_vnis  => ['5000-5010', '10000']
+    }
+
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\d+)$/,
+      [
+        [:topo_id, identity]
+      ],
+    ]
+    patterns
+  end
+
+  newparam(:topo_id, namevar: true) do
+    desc 'ID of the fabricpath topology. Valid values are integer in the range
+          1-63. Value of 0 is reserved for default topology.'
```

Comment:
Same as above comment. These haven't changed for a while now and not likely to change.